rust/routee-compass-core/src/model/unit/energy_unit.rs

```diff
@@ -1,66 +1,165 @@
+        const kWh2Kj: f64 = 3_600.;
+        const Kj2Kwh: f64 = 1. / 3600.;
+        const kWh2BTU: f64 = 3_412.14;
+        const BTU2Kwh: f64 = 1. / 3412.14;
+
         let conversion_factor = match (self, to) {
-            (S::GallonsGasoline, S::GallonsGasoline) => None,
-            (S::GallonsGasoline, S::KilowattHours) => Some(32.26),
-            (S::GallonsGasoline, S::LitersGasoline) => Some(3.78541),
-            (S::GallonsGasoline, S::LitersDiesel) => Some(0.866 * 3.78541),
-            (S::KilowattHours, S::GallonsGasoline) => Some(0.031),
-            (S::KilowattHours, S::KilowattHours) => None,
-            (S::KilowattHours, S::LitersGasoline) => Some(0.031 * 3.78541),
-            (S::KilowattHours, S::LitersDiesel) => Some(0.02457 * 3.78541),
-            (S::GallonsDiesel, S::GallonsDiesel) => None,
-            (S::GallonsDiesel, S::KilowattHours) => Some(40.7),
-            (S::GallonsDiesel, S::LitersGasoline) => Some(1.155 * 3.78541),
-            (S::GallonsDiesel, S::LitersDiesel) => Some(3.78541),
-            (S::KilowattHours, S::GallonsDiesel) => Some(0.02457),
-            (S::GallonsDiesel, S::GallonsGasoline) => Some(1.155),
-            (S::GallonsGasoline, S::GallonsDiesel) => Some(0.866),
-            (S::LitersGasoline, S::LitersGasoline) => None,
-            (S::LitersGasoline, S::LitersDiesel) => Some(0.866),
-            (S::LitersGasoline, S::GallonsGasoline) => Some(0.264172),
-            (S::LitersGasoline, S::GallonsDiesel) => Some(0.264172 * 0.866),
-            (S::LitersGasoline, S::KilowattHours) => Some(0.264172 * 32.26),
-            (S::LitersDiesel, S::LitersDiesel) => None,
-            (S::LitersDiesel, S::LitersGasoline) => Some(1.155),
-            (S::LitersDiesel, S::GallonsGasoline) => Some(0.264172 * 1.155),
-            (S::LitersDiesel, S::GallonsDiesel) => Some(0.264172),
-            (S::LitersDiesel, S::KilowattHours) => Some(0.264172 * 40.7),
-            (S::GallonsGasoline, S::GallonsGasolineEquivalent) => None,
-            (S::GallonsDiesel, S::GallonsGasolineEquivalent) => Some(1.14),
-            (S::KilowattHours, S::GallonsGasolineEquivalent) => Some(0.03),
-            (S::LitersGasoline, S::GallonsGasolineEquivalent) => Some(0.264172),
-            (S::LitersDiesel, S::GallonsGasolineEquivalent) => Some(3.78541 * 1.14),
-            (S::GallonsGasolineEquivalent, S::GallonsGasoline) => None,
-            (S::GallonsGasolineEquivalent, S::GallonsDiesel) => Some(0.8771929825),
-            (S::GallonsGasolineEquivalent, S::KilowattHours) => Some(33.3333333333),
-            (S::GallonsGasolineEquivalent, S::LitersGasoline) => todo!(),
-            (S::GallonsGasolineEquivalent, S::LitersDiesel) => todo!(),
-            (S::GallonsGasolineEquivalent, S::GallonsGasolineEquivalent) => None,
+            // Variants that do not need transformation
+            (x, y) if x == y => None,
+            (EU::Gasoline(V::GallonsUs), EU::GallonsGasolineEquivalent)
```

Comment:
we could probably reduce a lot of these match cases to a single line, for readability, using aliases like so:

```rust
impl EnergyUnit {
    #[allow(non_upper_case_globals)]
    const kwH: Self = Self::KilowattHours;
    const GDE: Self = Self::GallonsDieselEquivalent;
    const GGE: Self = Self::GallonsGasolineEquivalent;
}
```